synopsys-task/src/synopsys-task/tools-parameter.ts

```diff
@@ -644,6 +648,116 @@ export class SynopsysToolsParameter {
     return blackDuckFixPrData;
   }
 
+  async getFormattedCommandForSrm(): Promise<string> {
+    let command = "";
+    const assessmentTypeArray: string[] = [];
+    const assessmentTypes = inputs.SRM_ASSESSMENT_TYPES;
+    if (assessmentTypes != null && assessmentTypes.length > 0) {
+      for (const assessmentType of assessmentTypes) {
+        const regEx = new RegExp("^[a-zA-Z]+$");
+        if (
+          assessmentType.trim().length > 0 &&
+          regEx.test(assessmentType.trim())
+        ) {
+          assessmentTypeArray.push(assessmentType.trim());
+        } else {
+          throw new Error(
+            "Invalid value for "
+              .concat(constants.SRM_ASSESSMENT_TYPES_KEY)
+              .concat(constants.SPACE)
+              .concat(ErrorCode.INVALID_SRM_ASSESSMENT_TYPES.toString())
+          );
+        }
+      }
+    }
+
+    const srmData: InputData<Srm> = {
+      data: {
+        srm: {
+          url: inputs.SRM_URL,
+          apikey: inputs.SRM_APIKEY,
+          assessment: {
+            types: assessmentTypeArray,
+          },
+        },
+      },
+    };
+
+    if (inputs.SRM_BRANCH_NAME || inputs.SRM_BRANCH_PARENT) {
+      srmData.data.srm.branch = {
+        ...(inputs.SRM_BRANCH_NAME && { name: inputs.SRM_BRANCH_NAME }),
+        ...(inputs.SRM_BRANCH_PARENT && { parent: inputs.SRM_BRANCH_PARENT }),
+      };
+    }
+    if (inputs.SRM_PROJECT_NAME || inputs.SRM_PROJECT_ID) {
+      srmData.data.srm.project = {
+        ...(inputs.SRM_PROJECT_NAME && { name: inputs.SRM_PROJECT_NAME }),
+        ...(inputs.SRM_PROJECT_ID && { id: inputs.SRM_PROJECT_ID }),
+      };
+    } else {
+      const azureRepositoryName = this.getAzureRepositoryName();
+      taskLib.debug(`SRM project name: ${azureRepositoryName}`);
+      srmData.data.srm.project = {
+        name: azureRepositoryName,
+      };
+    }
+    if (inputs.BLACKDUCK_EXECUTION_PATH) {
+      srmData.data.blackduck = {
+        execution: {
+          path: inputs.BLACKDUCK_EXECUTION_PATH,
+        },
+      };
+    }
+
+    if (inputs.COVERITY_EXECUTION_PATH) {
+      srmData.data.coverity = {
+        execution: {
+          path: inputs.COVERITY_EXECUTION_PATH,
+        },
+      };
+    }
+    if (inputs.SRM_PROJECT_DIRECTORY) {
+      srmData.data.project = {
+        directory: inputs.SRM_PROJECT_DIRECTORY,
+      };
+    }
+
+    // Set Coverity or Blackduck Arbitrary Arguments
+    const coverityArgs = this.setCoverityArbitraryArgs();
+    const blackduckArgs = this.setBlackDuckArbitraryArgs();
+
+    if (Object.keys(coverityArgs).length > 0) {
+      srmData.data.coverity = { ...srmData.data.coverity, ...coverityArgs };
+    }
+    if (Object.keys(blackduckArgs).length > 0) {
+      srmData.data.blackduck = { ...srmData.data.blackduck, ...blackduckArgs };
+    }
+
+    const inputJson = JSON.stringify(srmData);
```

Comment:
to filter the empty data use the method `filterEmptyData()` before preparing json.

```
// Remove empty data from json object
srmData = filterEmptyData(srmdata);
```